File: 7-kyu/convert-color-image-to-greyscale/javascript/solution.js
const color2grey = (image) => {
  const newImage = [];
  for (const row of image) {
    const newRow = [];
    for (const pixel of row) {
      const average = Math.round(pixel.reduce((acc, nxt) => acc + nxt, 0) / pixel.length);
      const newPixel = [average, average, average];
      newRow.push(newPixel);
    }
    newImage.push(newRow);
  }
  return newImage;
};

File: 7-kyu/convert-color-image-to-greyscale/python/solution.py
def color_2_grey(colors):
    avg = lambda *args: round(sum(args) / len(args))
    greyscale = []
    for row in colors:
        new_row = []
        for pixel in row:
            new_pixel = [avg(*pixel)] * 3
            new_row.append(new_pixel)
        greyscale.append(new_row)
    return greyscale
